tests/Operation/WatchFunctionalTest.php

```diff
@@ -322,6 +322,70 @@ public function testNoChangeAfterResumeBeforeInsert()
         $this->assertMatchesDocument($expectedResult, $changeStream->current());
     }
 
+    public function testResumeTokenIsUpdatedAfterResuming()
+    {
+        $this->insertDocument(['_id' => 1]);
+
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $this->defaultOptions);
+        $changeStream = $operation->execute($this->getPrimaryServer());
+
+        $changeStream->rewind();
+        $this->assertNull($changeStream->current());
+
+        $this->insertDocument(['_id' => 2]);
+
+        $changeStream->next();
+        $this->assertTrue($changeStream->valid());
+
+        $expectedResult = [
+            '_id' => $changeStream->current()->_id,
+            'operationType' => 'insert',
+            'fullDocument' => ['_id' => 2],
+            'ns' => ['db' => $this->getDatabaseName(), 'coll' => $this->getCollectionName()],
+            'documentKey' => ['_id' => 2],
+        ];
+
+        $this->assertMatchesDocument($expectedResult, $changeStream->current());
+
+        $this->killChangeStreamCursor($changeStream);
+
+        $this->insertDocument(['_id' => 3]);
+
+        $changeStream->next();
+        $this->assertTrue($changeStream->valid());
+
+        $expectedResult = [
+            '_id' => $changeStream->current()->_id,
+            'operationType' => 'insert',
+            'fullDocument' => ['_id' => 3],
+            'ns' => ['db' => $this->getDatabaseName(), 'coll' => $this->getCollectionName()],
+            'documentKey' => ['_id' => 3],
+        ];
+
+        $this->assertMatchesDocument($expectedResult, $changeStream->current());
+
+        /* Triggering a consecutive failure will allow us to test whether the
+         * resume token was properly updated after the last resume. If the
+         * resume token updated, the next result will be {_id: 4}; otherwise,
+         * we'll see {_id: 3} returned again. */
```

Comment:
This is basically a manual prose version of the spec test for [PHPLIB-416](https://jira.mongodb.org/browse/PHPLIB-416)  and [DRIVERS-620](https://jira.mongodb.org/browse/DRIVERS-620), which is currently skipped in the `master` branch (see: [here](https://github.com/mongodb/mongo-php-library/blob/770e2afd49c04195fe1181c1eb419ef9b792a3ba/tests/SpecTests/ChangeStreamsSpecTest.php#L24)).

Reply:
Just to check, this failed before the `onIteration()` change?

Reply:
Correct. If I remove the call to `onIteration(false)` from `resume()`, `testResumeTokenIsUpdatedAfterResuming` fails and we encounter the error with duplicate events being returned as described in [PHPLIB-442](https://jira.mongodb.org/browse/PHPLIB-442).